source/modulo_core/src/Cell.cpp

```diff
@@ -419,6 +425,28 @@ bool Cell::on_cleanup() {
 
 rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn Cell::on_shutdown(const rclcpp_lifecycle::State& state) {
   RCUTILS_LOG_INFO_NAMED(get_name(), "on_shutdown() is called from state %s.", state.label().c_str());
+  uint8_t current_state = state.id();
```

Comment:
I might be wrong here but it seems to me that the `lifecycle_msgs::msg::State` is an enum (with uint8 type). Wouldn't it be nicer to create the variable `current_state` as that enum instead of the underlying type?

Reply:
lifecycle_msgs::msg::State is not an enum but rather a `static constexpr` that defines the symbol (e.g. `lifecycle_msgs::msg::State::PRIMARY_STATE_UNKNOWN` to have a value `uint8_t 0`

Reply:
okay nevermind then